qutebrowser/keyinput/modeparsers.py

```diff
@@ -189,7 +189,7 @@ def _handle_special_key(self, e):
                 return True
             else:
                 return super()._handle_special_key(e)
-        elif config.get('hints', 'mode') != 'number':
+        elif hintmanager.current_mode() != 'number':
```

Comment:
And IMO it should not be part of context - it's a config value that rarely changes.